plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/hms/TrinoHiveCatalog.java

```diff
@@ -261,6 +268,38 @@ public Transaction newCreateTableTransaction(
                 isUsingSystemSecurity ? Optional.empty() : Optional.of(session.getUser()));
     }
 
+    @Override
+    public void registerTable(ConnectorSession session, SchemaTableName schemaTableName, String metadataLocation)
+            throws TrinoException
+    {
+        if (tableExists(schemaTableName)) {
+            throw new TrinoException(TABLE_ALREADY_EXISTS, format("Table '%s' already exists", schemaTableName));
+        }
```

Comment:
Same as for Glue case -- if we remove this and have proper err handling, there is no question that the code is concurrently correct.

For HMS this is already test-covered `io.trino.plugin.iceberg.TestIcebergRegisterTableProcedure#testRegisterTableWithExistingTable`, you would just need to update the expected message, as it will be slightly different:

```
java.lang.AssertionError: 
Expecting message:
  <"Table already exists: 'tpch.test_register_table_with_existing_table_rwlb2koaqo'">
to match regex:
  <".*Table '(.*)' already exists.*">
but did not.
```

Reply:
Removed the check and modified the test case of Glue and HMS.